falcon/asgi/request.py

```diff
@@ -755,12 +756,29 @@ async def get_media(self, default_when_empty=_UNSET):
             self.options.default_media_type
         )
 
+        # PERF(kgriffs): Avoid using an additional await and flatten the call
+        #   stack when possible.
         try:
-            self._media = await handler.deserialize_async(
-                self.stream,
-                self.content_type,
-                self.content_length
+            use_shortcut = _shortcut_cache[handler]
```

Comment:
Maybe the whole shortcut cache could instead be a method returning the synchronous handler, or `None`?
And we would of course plop an `lru_cache` on top of it.

Reply:
That would be an option, but I think we would have to give up some cycles. With this simple local dict pattern I was aiming for the least overhead possible.

Reply:
Thoughts on ditching EAFP in favour of the below (CPython 3.8):
```python
>>> XML = object()
>>> SOAP = object()
>>> 
>>> _shortcut_cache1 = {XML: True, SOAP: False}
>>> 
>>> _shortcut_cache2 = {XML}
>>> 
>>> timeit.timeit(lambda: _shortcut_cache1[XML], number=int(1e8))
6.385023718999946
>>> timeit.timeit(lambda: _shortcut_cache1[SOAP], number=int(1e8))
6.587060416000895
>>>
>>> timeit.timeit(lambda: XML in _shortcut_cache2, number=int(1e8))
6.124332429000788
>>> timeit.timeit(lambda: SOAP in _shortcut_cache2, number=int(1e8))
6.186602156001754
```

Reply:
Maybe we could instead just have a private attribute on our own handlers?
And we would set it in the constructor, using `type(self) in our_handlers`, or just `type(self) is JSONHandler` etc.